lib/internal/loader/DefaultResolve.js

```diff
@@ -66,17 +67,22 @@ function resolve(specifier, parentURL) {
   }
 
   if (!preserveSymlinks) {
-    const real = realpathSync(internalURLModule.getPathFromURL(url), {
+    const real = realpathSync(getPathFromURL(url), {
       [internalFS.realpathCacheKey]: realpathCache
     });
     const old = url;
-    url = internalURLModule.getURLFromFilePath(real);
+    url = getURLFromFilePath(real);
     url.search = old.search;
     url.hash = old.hash;
   }
 
   const ext = extname(url.pathname);
-  return { url: `${url}`, format: extensionFormatMap[ext] || ext };
+
+  const format = extensionFormatMap[ext] || parentURL === undefined && 'cjs';
```

Comment:
This line is confusing IMO, I'd write it as either an if/else or at least use parenthesis.